kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala

```diff
@@ -172,6 +173,19 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
     super.stop()
   }
 
+  def getUserName(hs2ProxyUser: String): String = {
+    val sessionConf = Option(hs2ProxyUser).filter(_.nonEmpty).map(proxyUser =>
+      Map(KyuubiAuthenticationFactory.HS2_PROXY_USER -> proxyUser)).getOrElse(Map())
+
+    var userName: String = null
+    try {
+      userName = getUserName(sessionConf)
+    } catch {
+      case t: Throwable => throw new NotAllowedException(t.getMessage)
```

Comment:
sorry, I mean the below method, because this method invoke it.

```
  def getUserName(sessionConf: Map[String, String]): String = {
    // using the remote ip address instead of that in proxy http header for authentication
    val ipAddress = AuthenticationFilter.getUserIpAddress
    val realUser: String = ServiceUtils.getShortName(
      Option(AuthenticationFilter.getUserName).filter(_.nonEmpty).getOrElse("anonymous"))
    try {
    getProxyUser(sessionConf, ipAddress, realUser)
    } catch {
      case t: Throwable => throw new NotAllowedException(t.getMessage)
   }
  }
```